src/scripts/mod.rs

```diff
@@ -0,0 +1,309 @@
+//! Scripting engine to run scripts based on tags
+//!
+//!
+use anyhow::{anyhow, Result};
+use serde_derive::{Deserialize, Serialize};
+use std::fs::{self, File};
+use std::io::{self, prelude::*};
+use std::net::IpAddr;
+use std::path::PathBuf;
+use subprocess::{Exec, ExitStatus};
+use text_placeholder::Template;
+
+pub fn init_scripts() -> Vec<ScriptFile> {
```

Comment:
There is a common pattern in Rust and that is to initiate a struct resource through a method called `new`. So for this scenario we would obtain the scripts like we are already doing and inside `parse_scripts` we would call `ScriptFile::new()` that would contain the logic to parse this script. Let me know if this makes sense to you. 

```rust

```

Reply:
I implemented a `ScriptFile::new()` function, and moved the parsing logic to there. So the ScriptFile gets parsed and return an `Option`, and gets filtered on `Some()` and `None`. We can expand the information we give to the user in the future to be more clear if sommething goes wrong. Now it just gets discarded.